packages/cashier/src/pages/payment-agent/payment-agent-signup/country-selector/country-selector.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { ResidenceList, CountriesListResponse } from '@deriv/api-types';
+import { Formik, Field, FieldProps } from 'formik';
+import { Autocomplete, DesktopWrapper, MobileWrapper, SelectNative, Loading } from '@deriv/components';
+import { localize } from '@deriv/translations';
+import { useStore } from '../../../../hooks';
+import { TReactChangeEvent } from 'Types';
+import { observer } from 'mobx-react';
+
+type TCountrySelectorProps = {
+    selectedCountry: (country: ResidenceList[0]) => void;
+    className: string;
+};
+
+type TValues = {
+    country_input: string;
+};
+
+const CountrySelector = ({ selectedCountry, className }: TCountrySelectorProps) => {
+    const { client } = useStore();
+    const { fetchResidenceList } = client;
+    const [residence_list, setResidenceList] = React.useState<ResidenceList>();
+
+    const initial_form_values = {
+        country_input: '',
+    };
+
+    const validateFields = (values: TValues) => {
+        const errors: TValues = {
+            country_input: '',
+        };
+        const { country_input } = values;
+
+        if (!country_input) {
+            errors.country_input = localize('Please select the country of document issuance.');
+        } else if (!residence_list?.find(c => c.text === country_input)) {
+            errors.country_input = localize('Please select a valid country of document issuance.');
+        }
+
+        return errors;
+    };
+
+    const submitCountry = (values: TValues) => {
+        const matching_country = residence_list?.find(c => c.text === values.country_input);
+        if (matching_country) {
+            selectedCountry(matching_country);
+        }
+    };
+
+    React.useEffect(() => {
+        fetchResidenceList?.().then((response: CountriesListResponse) => {
+            setResidenceList(response.residence_list);
+        });
+    }, [fetchResidenceList]);
+
+    return (
+        <Formik initialValues={initial_form_values} validate={validateFields} onSubmit={submitCountry}>
+            {({ errors, handleBlur, handleChange, setFieldValue, touched, values }) => (
+                <div className={className}>
+                    {residence_list && residence_list.length ? (
+                        <fieldset className='proof-of-identity__fieldset'>
+                            <Field name='country_input'>
+                                {({ field }: FieldProps<string>) => (
+                                    <React.Fragment>
+                                        <DesktopWrapper>
+                                            <Autocomplete
```

Comment:
@farzin-deriv Yes, Handling this logic inside the component is better to prevent duplication. I will raise this inside the channel and we can have a card for this if all agree.